zrml/orderbook-v1/src/tests.rs

```diff
@@ -60,55 +68,233 @@ fn it_makes_orders() {
+                outcome_asset_amount: 300,
+                base_asset_amount: 1500,
+            }
+        );
+
+        let reserved_bob = Balances::reserved_balance(BOB);
+        // 5000 - (700 shares * 500 price) = 1500
+        assert_eq!(reserved_bob, 1500);
+
+        System::assert_last_event(
+            Event::<Runtime>::OrderFilled {
+                order_hash,
+                maker: BOB,
+                taker: ALICE,
+                filled: 700,
+                unfilled_outcome_asset_amount: 300,
+                unfilled_base_asset_amount: 1500,
+            }
+            .into(),
+        );
+
+        let alice_bal = <Balances as Currency<AccountIdTest>>::free_balance(&ALICE);
+        let alice_shares = Tokens::free_balance(outcome_asset, &ALICE);
+        assert_eq!(alice_bal, BASE + 3500);
+        assert_eq!(alice_shares, 300);
+
+        let bob_bal = <Balances as Currency<AccountIdTest>>::free_balance(&BOB);
+        let bob_shares = Tokens::free_balance(outcome_asset, &BOB);
+        // 3500 of base_asset lost, 1500 of base_asset reserved
+        assert_eq!(bob_bal, BASE - 5000);
+        assert_eq!(bob_shares, 700);
+    });
+}
+
 #[test]
 fn it_cancels_orders() {
     ExtBuilder::default().build().execute_with(|| {
+        let market_id = 0u128;
+        let market = market_mock::<Runtime>();
+        Markets::<Runtime>::insert(market_id, market);
+
         // Make an order from Alice to buy shares.
         let share_id = Asset::CategoricalOutcome(0, 2);
-        assert_ok!(Orderbook::make_order(
+        assert_ok!(Orderbook::place_order(
             RuntimeOrigin::signed(ALICE),
+            market_id,
             share_id,
             OrderSide::Bid,
             25,
             10
         ));
 
-        let order_hash = Orderbook::order_hash(&ALICE, share_id, 0);
+        let order_id = 0u128;
+        let order_hash = Orderbook::order_hash(&ALICE, order_id);
+
+        System::assert_last_event(
+            Event::<Runtime>::OrderPlaced {
+                order_hash,
+                order_id: 0,
+                order: Order {
+                    market_id,
+                    order_id,
+                    side: OrderSide::Bid,
+                    maker: ALICE,
+                    outcome_asset: share_id,
+                    base_asset: Asset::Ztg,
+                    outcome_asset_amount: 25,
+                    base_asset_amount: 10,
+                },
+            }
+            .into(),
+        );
 
         assert_noop!(
-            Orderbook::cancel_order(RuntimeOrigin::signed(BOB), share_id, order_hash),
+            Orderbook::remove_order(RuntimeOrigin::signed(BOB), order_hash),
             Error::<Runtime>::NotOrderCreator,
         );
 
-        assert_ok!(Orderbook::cancel_order(RuntimeOrigin::signed(ALICE), share_id, order_hash));
+        assert_ok!(Orderbook::remove_order(RuntimeOrigin::signed(ALICE), order_hash));
     });
```

Comment:
This test doesn't verify that the order is actually removed from `Orders`, not that assets are correctly unreserved. This needs to be fixed before we can deploy this on mainnet.

Reply:
e0ef7df9807cda708f4aac6419d17332f7631695

Reply:
The test should also verify that assets are correctly unreserved.